### PR TITLE
[ISSUE #3952]Optimize EventMeshServer#EventMeshServer method

### LINKED ISSUE #3952: [Enhancement] Optimize EventMeshServer#EventMeshServer method
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

Optimize EventMeshServer#EventMeshServer method. 

image

use a switch statement instead of multiple if statements to check for the server protocols

### Describe the solution you'd like

 use a switch statement instead of multiple if statements to check for the server protocols

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by mxsm**
Fixes #3952 

### Motivation

*Explain the content here.*
*Explain why you want to make the changes and what problem you're trying to solve.*

### Modifications

*Optimize EventMeshServer#EventMeshServer method*

### Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
- If a feature is not applicable for documentation, explain why?
- If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Review by Alonexc — APPROVED** [review]
LGTM

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/boot/EventMeshServer.java [hunk 1]
```diff
@@ -73,14 +73,16 @@ public EventMeshServer() {
...
         for (final String provideServerProtocol : provideServerProtocols) {
-            if (ConfigurationContextUtil.HTTP.equals(provideServerProtocol)) {
-                BOOTSTRAP_LIST.add(new EventMeshHttpBootstrap(this));
-            }
-            if (ConfigurationContextUtil.TCP.equals(provideServerProtocol)) {
-                BOOTSTRAP_LIST.add(new EventMeshTcpBootstrap(this));
-            }
-            if (ConfigurationContextUtil.GRPC.equals(provideServerProtocol)) {
-                BOOTSTRAP_LIST.add(new EventMeshGrpcBootstrap(this));
+            switch (provideServerProtocol) {
```
Comment (xwm1992) [inline eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/boot/EventMeshServer.java:76]: if you change to switch, you need add default case.

**Review by xwm1992 — CHANGES_REQUESTED** [review]

File: eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/boot/EventMeshServer.java (hunk 1 above)
Comment (mxsm) [reply eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/boot/EventMeshServer.java:80]: > if you change to switch, you need add default case.

I hava fix it and resubmit again

Comment (xwm1992) [reply eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/boot/EventMeshServer.java:80]: you need add default in switch condition. I think you just need to add:
switch (condition) {
 case 1:
 case 2:
default:
    break;
}

**Review by xwm1992 — CHANGES_REQUESTED** [review]

**Review by lrhkobe — APPROVED** [review]

**Pull request merged**